internal/jujuapi/httpproxy.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2024 Canonical.
+package jujuapi
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"github.com/juju/names/v5"
+	"github.com/juju/zaputil/zapctx"
+	"go.uber.org/zap"
+
+	"github.com/canonical/jimm/v3/internal/errors"
+	"github.com/canonical/jimm/v3/internal/rpc"
+)
+
+type httpProxier struct {
+	jimm JIMM // interface
+}
+
+func (s *httpProxier) Authenticate(ctx context.Context, w http.ResponseWriter, req *http.Request) error {
+	// extract auth token
```

Comment:
This could be 2 middlewares, one is authenticate, one is authorise. This does too much and is kinda misleading, you're checking permissions and specifically model permissions...